src/main/protobuf/remote_scrubbing.proto

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package remote_scrubbing;
+
+option java_package = "com.google.devtools.build.lib.remote";
+
+// Describes when and how to scrub remote cache keys.
+// See the documentation for the --experimental_remote_scrubbing_config flag.
+message Config {
+  // Describes a set of actions. An action must pass all criteria to match.
+  message Matcher {
+    // A regex matching the canonical label of the action owner.
```

Comment:
We probably want to warn user to be careful and avoid costly regex definition? This code will be invoked probably 100k-1M times on large codebase like ours.

Reply:
Added a warning.

(FWIW, an earlier version of this PR included an optimization to merge several `omitted_inputs` regexes into a single one, but I ended up throwing it away. We can still add it back if it makes a noticeable difference.